functions/src/metrics/logFeedback/index.ts

```diff
@@ -1,40 +1,18 @@
-import * as yup from 'yup';
 import validator from 'koa-yup-validator';
 import {createMetricsRouter} from '../../lib/routers';
 import * as feedbackController from '../../controllers/feedback';
-import {SessionMode, SessionType} from '../../../../shared/src/schemas/Session';
+import {CreateFeedbackBodySchema} from '../../../../shared/src/schemas/Feedback';
+import {CreateFeedbackBody} from '../../../../shared/src/types/Feedback';
 
 const router = createMetricsRouter();
 
-const feedbackBodySchema = yup.object({
-  exerciseId: yup.string().required(),
-  completed: yup.boolean().required(),
-  sessionId: yup.string().required(),
-  host: yup.boolean(),
-
-  question: yup.string().required(),
-  answer: yup.boolean().required(),
-  comment: yup.string(),
-
-  sessionType: yup
-    .mixed<SessionType>()
-    .oneOf(Object.values(SessionType))
-    .required(),
-  sessionMode: yup
-    .mixed<SessionMode>()
-    .oneOf(Object.values(SessionMode))
-    .required(),
-});
-
-type Body = yup.InferType<typeof feedbackBodySchema>;
-
 export const feedbackRouter = router.post(
   '/',
   validator({
-    body: feedbackBodySchema,
+    body: CreateFeedbackBodySchema,
   }),
   async ({request, response}) => {
-    const feedback = request.body as Body;
+    const feedback = request.body as CreateFeedbackBody;
```

Comment:
Destruct `body` instead of `request.body` and it should give you the correct type instead :)

Reply:
<img width="384" alt="Screenshot 2023-09-21 at 15 23 56" src="https://github.com/29ki/29k/assets/7523828/5bc4904a-e2a3-43ef-b623-e9b0f2ae1520">

Reply:
seems like we use `request.body as ...`  across the codebase?
<img width="377" alt="Screenshot 2023-09-21 at 15 25 04" src="https://github.com/29ki/29k/assets/7523828/08619ce2-317a-4344-9a72-a2f72d2e3b3e">

Reply:
Something off with the validation lib... resolving for now